nimbus/src/evaluator.rs

```diff
@@ -362,10 +397,70 @@ mod tests {
             &available_randomization_units,
             &Default::default(),
             &experiment2,
+            &schema_version,
         )
         .unwrap();
         assert!(
             matches!(enrollment.status, EnrollmentStatus::Enrolled { reason: EnrolledReason::Qualified, .. })
         );
     }
+
+    #[test]
+    fn test_schema_version_validation() {
+        let experiment1 = Experiment::V1(V1_0_0 {
+            schema_version: "1.0.0".to_string(),
+            slug: "TEST_EXP1".to_string(),
+            is_enrollment_paused: false,
+            bucket_config: BucketConfig {
+                randomization_unit: RandomizationUnit::NimbusId,
+                namespace: "bug-1637316-message-aboutwelcome-pull-factor-reinforcement-76-rel-release-76-77".to_string(),
+                start: 0,
+                count: 2000,
+                total: 10000,
+            },
+            branches: vec![
+                Branch {slug: "control".to_string(), ratio: 1, feature: None },
+                Branch {slug: "blue".to_string(), ratio: 1, feature: None }
+            ],
+            reference_branch: Some("control".to_string()),
+            ..Default::default()
+        });
+
+        // Create a second experiment with a non-matching schema_version
+        let experiment2 = match experiment1.clone() {
+            Experiment::V1(mut e) => {
+                e.schema_version = "2.0.0".to_string();
+                Experiment::V1(e)
```

Comment:
This test is a neat example of where my mental model is getting tripped up - something about creating an `Experiment::V1` struct that has `schema_version: "2.0.0"` seems weird to me, and like the checking is happening in the wrong place. I would expect us to have produced an `Experiment::V2` struct for something with `schema_version: "2.0.0"`, or to have ignored it completely.

(That's not to say that we shouldn't do it this way! Just noting that I found it confusing, and may be worth some additional docs or context-setting or something).